src/test/java/io/neonbee/test/base/ODataRequest.java

```diff
@@ -101,20 +101,22 @@ public ODataRequest setBody(Buffer body) {
      */
     public Future<HttpResponse<Buffer>> send(NeonBee neonBee) {
         Vertx vertx = neonBee.getVertx();
-        DeploymentOptions opts = new DeploymentOptions(readConfigBlocking(vertx, ServerVerticle.class.getName()));
-        int port = opts.getConfig().getInteger(CONFIG_PROPERTY_PORT_KEY, -1);
-        String basePath = Optional.ofNullable(opts.getConfig().getJsonObject("odata"))
-                .map(odata -> odata.getString("basePath")).orElse(DEFAULT_ODATA_BASE_PATH);
-
-        WebClientOptions clientOpts = new WebClientOptions().setDefaultHost("localhost").setDefaultPort(port);
-        HttpRequest<Buffer> httpRequest = WebClient.create(vertx, clientOpts).request(method, basePath + getUri());
-
-        httpRequest.queryParams().addAll(query);
-        httpRequest.putHeaders(headers);
-        Optional.ofNullable(interceptor).ifPresent(i -> i.accept(httpRequest));
-        httpRequest.queryParams(); // ensures that query params are encoded
-
-        return Optional.ofNullable(body).map(b -> httpRequest.sendBuffer(b)).orElse(httpRequest.send());
+        return readConfig(vertx, ServerVerticle.class.getName()).compose(config -> {
+            DeploymentOptions opts = new DeploymentOptions(config);
+            int port = opts.getConfig().getInteger(CONFIG_PROPERTY_PORT_KEY, -1);
+            String basePath = Optional.ofNullable(opts.getConfig().getJsonObject("odata"))
+                    .map(odata -> odata.getString("basePath")).orElse(DEFAULT_ODATA_BASE_PATH);
+
+            WebClientOptions clientOpts = new WebClientOptions().setDefaultHost("localhost").setDefaultPort(port);
+            HttpRequest<Buffer> httpRequest = WebClient.create(vertx, clientOpts).request(method, basePath + getUri());
+
+            httpRequest.queryParams().addAll(query);
+            httpRequest.putHeaders(headers);
+            Optional.ofNullable(interceptor).ifPresent(i -> i.accept(httpRequest));
+            httpRequest.queryParams(); // ensures that query params are encoded
+
+            return Optional.ofNullable(body).map(b -> httpRequest.sendBuffer(b)).orElse(httpRequest.send());
```

Comment:
why not with method reference
`return Optional.ofNullable(body).map(httpRequest::sendBuffer).orElse(httpRequest.send());`